include/deal.II/lac/la_parallel_vector.h

```diff
@@ -1325,56 +1383,144 @@ namespace LinearAlgebra
 
 #ifndef DOXYGEN
 
+    namespace internal
+    {
+      template <typename Number, typename MemorySpace>
+      struct la_parallel_vector_functions
+      {
+        static inline void
+        memory_space_assert()
+        {
+          static_assert(std::is_same<MemorySpace, Host>::value ||
+                          std::is_same<MemorySpace, CUDA>::value,
+                        "MemorySpace should be Host or CUDA");
```

Comment:
couldn't you just move the `static_assert` to class-level? The function here is not called anywhere, and the error comes out at compile-time, so a call would not be useful to begin with.